app/src/main/java/net/programmierecke/radiodroid2/adapters/ItemAdapterIconOnlyStation.java

```diff
@@ -0,0 +1,82 @@
+package net.programmierecke.radiodroid2.adapters;
+
+import android.content.SharedPreferences;
+import android.util.TypedValue;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.fragment.app.FragmentActivity;
+import androidx.preference.PreferenceManager;
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.RecyclerView;
+
+import net.programmierecke.radiodroid2.PlayerServiceUtil;
+import net.programmierecke.radiodroid2.R;
+import net.programmierecke.radiodroid2.Utils;
+import net.programmierecke.radiodroid2.data.DataRadioStation;
+import net.programmierecke.radiodroid2.utils.RecyclerItemMoveAndSwipeHelper;
+import net.programmierecke.radiodroid2.utils.SwipeableViewHolder;
+
+public class ItemAdapterIconOnlyStation extends ItemAdapterStation {
+
+    private final String TAG = "AdapterIconOnlyStations";
+
+    class StationViewHolder extends ItemAdapterStation.StationViewHolder implements View.OnClickListener, SwipeableViewHolder {
```

Comment:
Is it required to extend `ItemAdapterStation.StationViewHolder`, couldn't they both just extend some base class? Just don't want to bring such dependence if possible.

Reply:
> Is it required to extend `ItemAdapterStation.StationViewHolder`, couldn't they both just extend some base class? Just don't want to bring such dependence if possible.

Not liking such side-dependencies either, I considered that, too, but that common super class would have to be yet another abstract class then, as on itself, it would be unusable. I thought that in this case `ItemAdapterIconOnlyStation` really is  a specialization of `ItemAdapterStation` – with  the special circumstance that it has less functionality. So I thought that all in all solution extending  `ItemAdapterStation` would have less potential for confusion and would be less evil than the alternative. But I may have overlooked something.

Reply:
Ok, it's not a big deal and always could be resolved later.